v3.2

AngRodrigues opened the pull request:
## Description

This version is compatible with LPF version available at https://github.com/Loop3D/LoopProjectFile/pull/42

### Includes features described in:
PR - https://github.com/Loop3D/map2loop/pull/140
PR - https://github.com/Loop3D/map2loop/pull/154
PR - https://github.com/Loop3D/map2loop/pull/157
PR - https://github.com/Loop3D/map2loop/pull/158
 PR - https://github.com/Loop3D/map2loop/pull/159

## Major changes:

1. **Merged thickness calculator outputs:**
Methods `set_thickness_calculator` and `get_thickness_calculator` still available
User can now choose up to 5 thickness calculators; also works with user-defined ThicknessCalculator class. 
Default is `InterpolatedStructure`. 
LPF stores results and name of the thickness calculator as a new field - `LoopProjectFile.thicknessCalculatorType`, which is a part of the `StratigraphicLog` object. 
`ThicknessCalculatorAlpha` updated to logic of the other calculators (e.g., to work from `sampled_basal_contacts`)

```
from map2loop.thickness_calculator import ThicknessCalculatorAlpha, InterpolatedStructure, StructuralPoint

[..set up the project..]

project.set_thickness_calculator([StructuralPoint(), InterpolatedStructure()])
project.get_thickness_calculator()

Output:
['StructuralPoint', 'InterpolatedStructure']
```

2. **Minimum fault length**
Minimum fault length is now a parameter of the config file, 
removes the faults with length under defined value; if set to `None`, minimum fault length is calculated from project's `bbox`, 
Removed the setter method to avoid double definition; getter method still available through `proj.map_data.get_minimum_fault_length()`

```
config_dictionary = {
  "structure" : {
    "dipdir_column": "azimuth2",
    "dip_column": "dip",},
  "geology" : {
    "unitname_column": "unitname",
    "alt_unitname_column": "code"}, 
  "fault": {
      "minimum_fault_length": 0.0}
}
```


3. **added `all_basal_contacts` object to mapdata**
`mapdata.all_basal_contacts` -> abnormal+basal contacts , 
`mapdata.basal_contacts` -> holds only the basal. 
Reconstruction of `sampled_basal_contacts` is based on `basal_contacts`

4. **Added folder with datasets in `map2loop/_datasets/geodata_files/hamersley`**
The idea behind this is to have a data loading function, e.g., at the moment `load_hamersley_geology()` returns a gdf with the geology; structure and dtm are available, and others will keep being added as needed. This is quite useful for tests, but should also be useful for when we implement https://github.com/Loop3D/map2loop/issues/74. The only thing about this is the location, which implies a long import: from map2loop._datasets.geodata_files.hamersley import load_hamersley_geology. Happy to change the file location - if anyone has a better suggestion?

5. **Added `DependencyChecker` class to `__init__.py`**
This was based on LG's suggestion to check the LPF version on import. 
I decided to expand on this and add this check for all libraries in `map2loop/dependencies.txt` 

6. **Added ignore codes both for lithology and faults**
works by feature name, and similar to the existing method. 
Added methods for set and get with `_ignore_lithology_codes()` and `_ignore_fault_codes()`. 
Example of working example code to achieve this can be seen in `map2loop/tests/project/test_ignore_codes_setters_getters.py`

## Minor changes:

- tmpfile is now used to create the localGdalfile
- changed to f"" strings where possible
- change from os.path.join to pathlib.Path where possible
- added geopandas to csv converter in mapdata (it was on a #TODO)
- added check and error message to make sure that all units in user_defined_stratigraphic_column are in the geology file (otherwise project creation fails; at least now user has a warning).
- renamed distance to stratigraphic_distance in basal_contacts creation
- simplified issue templates for easier form fill

**Added tests:**
- for some of the functions in mapdata.py (includes minimum_fault_length)
- for thickness_compute in project.py
- ignore_codes setters and getters

Fixes #86 
Fixes #110 
Fixes #111 
Fixes #12 
Fixes #143 
Fixes #137 
Fixes #119 
Fixes #155 

## Type of change

- [x] Documentation update
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Test improvement

## How Has This Been Tested?

Branches: 
- m2l - https://github.com/Loop3D/map2loop/tree/fix/thicknesses_AR_2
- lpf: https://github.com/Loop3D/LoopProjectFile/tree/fix_thicknesses

# 
:warning: **actions before merging!** :warning:

- [x] - release LPF 0.2
- [x] dependencies.txt need to be updated for the LPF version -> *0.2.1*
- [x] make actions same as LPF and m2m (maybe add tests if possible)
- [x] - actions need to be reversed to use LPF main branch. Search for @fix_thicknesses
- [x] - docs docker build needs to be also changed to remove the @fix_thicknesses branch
- [x] - add feat commit for version bump
- [x] pypi build & install tested and passed
- [x] test conda 
- [ ] add to release notes: set_ignore_codes has change to set_ignore_lithology_codes

lachlangrose committed:
fix: trigger build

lachlangrose committed:
style: style fixes by ruff and autoformatting by black

github-actions[bot] committed:
chore(master): release 3.1.1

AngRodrigues committed:
fix: update doc deploy

github-actions[bot] committed:
chore(master): release 3.1.2

AngRodrigues committed:
docs: remove lavavu; no need for 3D vis for the examples

AngRodrigues committed:
style: style fixes by ruff and autoformatting by black

AngRodrigues committed:
fix: remove tini from docker

github-actions[bot] committed:
chore(master): release 3.1.3

lachlangrose committed:
Update issue templates

lachlangrose committed:
Update issue templates

AngRodrigues committed:
feat: added issue_templates

AngRodrigues committed:
feat: added issue_templates

edit dddf70b chore: added issue_templates

AngRodrigues committed:
fix: add pull request template

AngRodrigues committed:
fix: add pull request template

AngRodrigues committed:
fix: few typos grammar

AngRodrigues committed:
fix: few typos grammar

AngRodrigues committed:
fix: tests should be with pytest

AngRodrigues committed:
fix: tests should be with pytest

AngRodrigues committed:
Merge pull request #75 from Loop3D/fix/adding_issue_templates

Update issue templates

AngRodrigues committed:
Revert "Update issue templates"

AngRodrigues committed:
Revert "Update issue templates"

rabii-chaarani committed:
Merge branch 'revert-75-fix/adding_issue_templates' of https://github.com/Loop3D/map2loop into revert-75-fix/adding_issue_templates

AngRodrigues committed:
fix: add issue templates back

AngRodrigues committed:
fix: correct grammar for clarity

AngRodrigues committed:
Merge pull request #83 from Loop3D/revert-75-fix/adding_issue_templates

Revert "Update issue templates" PR

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into v3.2

rabii-chaarani committed:
Merge branch 'master' into v3.2

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into v3.2

rabii-chaarani committed:
Merge branch 'master' into v3.2

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into v3.2

AngRodrigues committed:
fix: squashed commits of: - fixing the non-use of minimum_fault_length, - use tmpfile to create temporary files, - added random test for mapdata functions:get/set working projection, - added functionality and comments to basal_contacts_build

AngRodrigues committed:
tests: add more functionality

AngRodrigues committed:
fix: leave as is for now

AngRodrigues committed:
fix: finalise the thickness calculator outputs

RoyThomsonMonash committed:
fix: Remove hjson and rely on json only

AngRodrigues committed:
fix: update tests/ small fixes

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into remove_hjson

AngRodrigues committed:
fix: make sure dependencies are right

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into v3.2

AngRodrigues committed:
fix: linting issue

AngRodrigues committed:
fix: try again

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into fix/thicknesses_AR_2

AngRodrigues committed:
fix: update thickness tests//fix minor typos

AngRodrigues committed:
fix: correct some typos/update tests

AngRodrigues committed:
fix: typo in tests

AngRodrigues committed:
Update README.md

AngRodrigues committed:
fix: add the basal_contacts_abnormal_check

AngRodrigues committed:
fix: add error handles for json, url, filenotfound

AngRodrigues committed:
fix: remove typo

AngRodrigues committed:
fix: remove tqdm

AngRodrigues committed:
Merge pull request #109 from Loop3D/remove_hjson

fix: Remove hjson and rely on json only

rabii-chaarani committed:
fix: added check for collocated points

AngRodrigues committed:
fix: issue 122

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into fix/thicknesses_AR_2

AngRodrigues committed:
fix: upd gdal as per master

AngRodrigues committed:
fix: remove test from server

AngRodrigues committed:
fix: update str format

rabii-chaarani committed:
fix: now use DBSCAN to aggregate collocated poins

rabii-chaarani committed:
fix: removed addition

rabii-chaarani committed:
fix: added warning when collocated points detected

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into fix/thicknesses_AR_2

AngRodrigues committed:
fix: add str for paths - gdal does not accept pathlibs

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into fix/thicknesses_AR_2

AngRodrigues committed:
fix: update minimum_fault_length as per code review

AngRodrigues committed:
fix: add pathlibs where possible in m2model_wrapper

AngRodrigues committed:
fix: add pathlib in mapdata.py

AngRodrigues committed:
fix: merge coll points branch and remove linalg catch

AngRodrigues committed:
fix: revert pathlib in map2model.run

AngRodrigues committed:
fix: proj should not have temp files

AngRodrigues committed:
fix: update the minimum_fault_length

AngRodrigues committed:
tests: add tests for minimum_fault_length

AngRodrigues committed:
fix: remove redundancy from project.py

AngRodrigues committed:
fix: update calculate minimum_fault_length for 5% of the map area

AngRodrigues committed:
fix: linting

AngRodrigues committed:
fix: typo

AngRodrigues committed:
fix: remove repetitive print stat

AngRodrigues committed:
fix: adjust thickness_calc workflow for modularity

AngRodrigues committed:
fix: remove unnecessary fields

AngRodrigues committed:
fix: update thickness tests

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into fix/thicknesses_AR_2

AngRodrigues committed:
fix: add ignore_fault_codes and tests for ignore_lithology_code and ignore_fault_code

rabii-chaarani committed:
fix: try relative path ../_datasets

rabii-chaarani committed:
fix: use only path instead of pkg_resources

rabii-chaarani committed:
fix: try full path

rabii-chaarani committed:
fix: load only using path /hamersley

rabii-chaarani committed:
fix: try ./map2loop/_datasets/geodata_files/

rabii-chaarani committed:
fix: use correct path

rabii-chaarani committed:
refactor: back to original code

AngRodrigues committed:
fix: add the whole hamersley folder

AngRodrigues committed:
fix: update docker for docs

AngRodrigues committed:
fix: update the docs-deploy for master branch only

AngRodrigues committed:
fix: update CI to build on master only

AngRodrigues committed:
fix: remove redundancy

AngRodrigues committed:
fix: update the server test and handle timeout properly

AngRodrigues committed:
fix: updated ReadMe to have the right install steps (temporary measure)

AngRodrigues committed:
fix: update CI - test wheels on master only

AngRodrigues committed:
fix: add test to sdist build in CI to ensure tests are run in the non-master branches

AngRodrigues committed:
fix: update pytest in ci

AngRodrigues committed:
fix: revert ci

AngRodrigues committed:
fix: add pytest in the right place

AngRodrigues committed:
fix: update pytest location

AngRodrigues committed:
fix: manifest update

AngRodrigues committed:
fix: update ci again

AngRodrigues committed:
investigate

AngRodrigues committed:
find the install folder

AngRodrigues committed:
try again

AngRodrigues committed:
update ci

AngRodrigues committed:
fix: add checkout step to build sdist

AngRodrigues committed:
fix: run pytest with test sdist action

AngRodrigues committed:
fix: revert CI and let tests only run on main branch

AngRodrigues committed:
fix: add version check new class to check install of  dependencies.txt; includes version number check

AngRodrigues committed:
fix: organise manifest

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into v3.2

AngRodrigues committed:
Merge branch 'fix/thicknesses_AR_2' of https://github.com/Loop3D/map2loop into v3.2

AngRodrigues committed:
fix: linting fixes from merges

AngRodrigues committed:
Merge branch 'master' of https://github.com/Loop3D/map2loop into v3.2

AngRodrigues committed:
fix: update CI to run all jobs on branch

AngRodrigues committed:
fix: update map2model paths

AngRodrigues committed:
fix: update LPF to install from branch

AngRodrigues committed:
fix: update install from branch

AngRodrigues committed:
fix: update the lpf to branch in wheel testing

AngRodrigues committed:
fix: build the docs on this branch

AngRodrigues committed:
fix: keep docs separate

AngRodrigues committed:
fix: pypi and conda upld runs only on master

AngRodrigues committed:
fix: test the docs on the branch too

AngRodrigues committed:
fix: update docs docker to build lpf from branch

AngRodrigues committed:
fix: install sphinx with apt-get?

AngRodrigues committed:
docs: change sphinx build path

AngRodrigues committed:
fix: update build path docs

AngRodrigues committed:
docs: update build requirements

AngRodrigues committed:
fix: add beartype

AngRodrigues committed:
docs: typo in docker?

AngRodrigues committed:
fix: remove redundant beartpe install

AngRodrigues committed:
chore: typo

AngRodrigues committed:
fix: simplify issue templates

AngRodrigues committed:
chore: hjson not in 3.2

AngRodrigues committed:
fix: remove hjson from dependencies as not in 3.2

lachlangrose commented on .github/workflows/CD.yml:
```diff
@@ -75,7 +75,8 @@ jobs:
           pip install --upgrade pip
           pip install numpy==1.26.4
-          pip install geopandas shapely networkx owslib map2model loopprojectfile beartype gdal==3.8.4 hjson pytest scikit-learn
+          pip install geopandas shapely networkx owslib map2model beartype gdal==3.8.4 pytest scikit-learn
```
I think we may not need to pin gdal anymore, would be good to test if 3.8.4 is required or if we could just use >3.8

AngRodrigues replied:
seems to work, gdal is now unpinned from ci

lachlangrose commented on .github/workflows/CD.yml:
```diff
@@ -150,6 +151,8 @@ jobs:
           pip install --upgrade pip
           pip install numpy==1.26.4
           pip install -r dependencies.txt
+          pip uninstall loopprojectfile -y
```
This is just for the PR and will be removed when merging to master

AngRodrigues replied:
yep, there are notes at the bottom of the PR on what to change before merging into master. 
Will add them as tasks.

lachlangrose commented on docs/Dockerfile:
```diff
@@ -10,10 +10,11 @@ RUN apt-get update -qq && \
     libc-dev \
     make\
     libgl1\
+    python3-sphinx\
```
I think sphinx should be installed using pip/conda - it is meant to be in the dependencies.txt file in docs, which should be installed from the dependenciesdocs.txt file

AngRodrigues replied:
I had an error on docs build previously that this seemed to fix (even with sphinx installed through conda), but it looks like it's working now, so reverted this

lachlangrose commented on map2loop/config.py:
@AngRodrigues and @rabii-chaarani are we still planning on removing the legacy format for 3.2?

AngRodrigues replied:
@lachlangrose This is a feature that we ended up not working on given that other issues were raised in the meantime. 
I am keen to merge these bug fixes asap, and adding that feature would keep delaying the merge on this PR, so could we leave that for a future release?

I tested 3.2 quickly with the legacy files, and it seems to run the project fine.  The docs will also be changed to encourage users to use the config dictionary - see this branch in the notebooks -> https://github.com/Loop3D/map2loop-notebooks/tree/fix_thicknesses_update

lachlangrose commented on map2loop/deformation_history.py:
```diff
@@ -7,14 +7,16 @@
     Attributes
     ----------
-    minimum_fault_length_to_export: float
+    minimum_fault_length: float
```
We can remove this doc, it should be in config.

lachlangrose commented on map2loop/deformation_history.py:
```diff
@@ -286,8 +267,8 @@ def get_faults_for_export(self):
         """
-        logger.info("Getting faults for export")
-        return self.faults[self.faults["length"] >= self.minimum_fault_length_to_export].copy()
+        mfl = Config().fault_config["minimum_fault_length"]
```
This just gets the default value for the fault config, not the current mapdata config. I think we either need to pass a parameter to the deformation history that is min fault length, or we should  pass the current proj.map_data.config object to the deformation history. I cant decide what is the best solution. I am leaning towards just pass the config object becuase it'll be useful for the fold stuff in the future i imagine.

lachlangrose replied:
Or pass the project and let the project manage that.

AngRodrigues replied:
This has been changed and seems to be working alright now.

lachlangrose commented on map2loop/map2model_wrapper.py:
I will finish and merge the code to remove map2model, or at least provide an option to bipass it.

lachlangrose commented on map2loop/mapdata.py:
```diff
@@ -298,28 +295,66 @@ def get_colour_filename(self):
 
     @beartype.beartype
-    def set_ignore_codes(self, codes: list):
+    def set_ignore_lithology_codes(self, codes: list):
```
We need to put this in the release notes that the set_ignore_codes has change to set_ignore_lithology_codes

AngRodrigues replied:
added to the todo before merge list in the PR above

lachlangrose commented on map2loop/mapdata.py:
```diff
@@ -91,10 +89,9 @@ def __init__(self, tmp_path: str = "", verbose_level: VerboseLevel = VerboseLeve
-
         self.config = Config()
+        self.minimum_fault_length = None
 
```
I'm going to rmeove this and make a property accessor

lachlangrose commented on map2loop/project.py:
```diff
@@ -368,28 +379,7 @@ def get_sampler(self, datatype: Datatype):
-        return self.deformation_history.get_minimum_fault_length()
-
+    
     # Processing functions
```
I added this back but it could be removed, it all links to the minimum_fault_length in the config file though so we only keep track of one variable

lachlangrose commented on dependencies.txt:
```diff
@@ -5,9 +5,8 @@ shapely
-loopprojectfile==0.1.3
+loopprojectfile==0.1.4
 beartype
 gdal==3.8.4
```
We should test this

AngRodrigues replied:
The pinned version will actually have to be 0.1.5, assuming that the PR in https://github.com/Loop3D/LoopProjectFile/pull/42 is released as a new version

lachlangrose replied:
I meant the gdal requirement, not projectfile sorry

AngRodrigues replied:
The gdal requirement seems to be needed - see https://github.com/Loop3D/map2loop/pull/153/commits/b7c9c049393ba7eb0878be20f046413de5e18801 and subsequent commits. It's strange because it has worked in previous commits without a pin...

lachlangrose commented on dependencies.txt:
```diff
@@ -5,9 +5,8 @@ shapely
 beartype
 gdal==3.8.4
-hjson
 pytest
```
Do we want to have "testing" "dev" and "release" requirements?

AngRodrigues replied:
There might not be a big difference between the different requirements for map2loop, and we'd have to maintain more files. Is there a specific advantage in doing this that you're thinking of?

lachlangrose replied:
Using pyproject it is possible to have specific requirements for testing/development, I'm not sure how to do it using conda though. It isn't a big deal though

AngRodrigues replied:
I think it's a good idea for a new feature, will add this to the map2loop project, but perhaps we'll leave it out of this release.

lachlangrose committed:
doc: removing min fault length attribute docstring

lachlangrose committed:
fix: adding min fault length setter back to project. Linking to config attribute

lachlangrose committed:
fix: min fault length setter/getter for mapdata uses config attribute

lachlangrose committed:
No data min fault length is -1

AngRodrigues committed:
chore: test unpin gdal from ci

AngRodrigues committed:
fix: remove sphinx from docs docker build

AngRodrigues committed:
chore: try again

lachlangrose committed:
fix: map2model fault fault relationships (#140)

* fix: fault topology merge was using incorrect lookups

* fix: run map2model for user defined stratigraphic column

* removing logger from this pr

* fix: use fault id not Fault_{id}

lachlangrose committed:
feat: make map2model optional using geopandas spatial joins as a replacement (#154)

* fix: stratigraphic alpha sorter was inverted, reverse sorter

* ignore units without contact for max contact length sorter

* updated orientation sorter to use correct trigonometry for strike/dip lines.

also change to using all intersections along line not only the first intersection. perhaps this should be a different sorter?

* revert back to original orientation sorter

* style: black

* updating WA json to work

* remove unused argument documentation

* only run doc build on master

* use geopandas to produce topology graphs instead of map2model.

Basically just using sjoins and buffers. Returns the same for faults and contacts but
we have some differences for fault-unit intersections.

* update sorter to not use stratigraphic order hint and deprecate hint sorter

* update deformation history to use eventId not name for merging

* bipass old map2model run

* use property accessor to make sure that variables are up to date

* remove sorted units from sorter call

* removing comment and condaforge channel

* fix: remove map2model call from map2loop

this is a quick fix, and needs to be cleaned up as the tmp files for map2model are still being created
and the old code is just commented

* update networkx sorter to have a name2index map

* fix: add mode to choose betweeen m2m and geopandas

* remove gdal fix

* linting

* style: black formatting

AngRodrigues committed:
fix: make the minimum_fault_length updatable through project

AngRodrigues committed:
fix: add libgdal to docs docker

AngRodrigues committed:
typo

AngRodrigues committed:
fix: try other gdal on docs

AngRodrigues committed:
more gdal issues

AngRodrigues committed:
fix: revert the gdal pin removal to see if that's the issue

lachlangrose commented on map2loop/deformation_history.py:
```diff
@@ -29,14 +31,15 @@ class DeformationHistory:
         self.history = []
         self.fault_fault_relationships = []
-
+        self.minimum_fault_length = project.get_minimum_fault_length()
```
@AngRodrigues this still won't work. If the minimum fault length is set using the calculate_minimum_fault_length function, as in line 277 the minimum fault length will not be up to date with the project/config minimum fault length. 

The project needs to be set as an attribute of the deformation_history class that is referenced back to when updating the value either using a property accessor or directly using the project.set_minimum_fault_length.

lachlangrose replied:
It also calls get_minimum_fault_length() before reading the config file because its set when the deformationhistory object is created.

lachlangrose committed:
fix: use minimum fault length from project

lachlangrose committed:
fix: use separate thickness table in LPF & add active thickness to project (#157)

* fix: use separate thickness table in LPF

* docs: remove unused docstring

* fix: min fault length can be int

* fix: bug in output of StructuralPoint

* fix: add the active_thickness_flag

---------

Co-authored-by: AngRodrigues <angela.rodrigues@monash.edu>

AngRodrigues committed:
feat: do not accept any legacy files

AngRodrigues committed:
Merge branch 'v3.2' of https://github.com/Loop3D/map2loop into feat--remove-legacy_2

AngRodrigues committed:
chore: small typo

lachlangrose committed:
remove legacy format arguments

lachlangrose committed:
add config file path back

AngRodrigues committed:
fix: allow json files, check for old keys

AngRodrigues committed:
chore: add/fix/update logger outputs

AngRodrigues committed:
chore: more logging

AngRodrigues committed:
chore: update tests to not rely on warnings

AngRodrigues committed:
chore: proper string formatting

AngRodrigues committed:
fix: small fixes for deprecation/linting warnings in test sessions

AngRodrigues committed:
chore: update actions from deprecation

AngRodrigues committed:
fix: issue 155  (#158)

* fix: simple fix for #155

* fix: decimation factor allows floats

* fix: decimation int

lachlangrose committed:
Merge pull request #159 from Loop3D/feat--remove-legacy_2

feat: remove option for legacy hson files

AngRodrigues committed:
fix: remove print statement left behind

AngRodrigues committed:
chore: update docstrings

AngRodrigues committed:
fix: gdal: test if conda issue

AngRodrigues committed:
fix: revert

AngRodrigues committed:
remove gdal pin??

AngRodrigues committed:
fix: keep gdal pin for now

AngRodrigues committed:
fix: add debug info to thickness calculators (#161)

* fix: initial commit

* fix: add debug info and warning for bad calculations

* fix: add line length control to thickness calculators

* fix: refactor to avoid repetitive code

* fix: typo

* fix: remove list comprehension - wky

* fix list to df

* fix: revert to lst comprehension

* fix: make line length attribute of the TC class

* fix: typos

* fix: syntax

* fix: add location tracking

* fix: init commit to remove lst comprehension

AngRodrigues committed:
chore: empty commit to trigger ci run

AngRodrigues committed:
chore: update typing to avoid deprecation

AngRodrigues committed:
fix: update ubuntu

AngRodrigues committed:
ci: small updates

AngRodrigues committed:
fix: dependencies updated

AngRodrigues committed:
fix: update LPF dependency

AngRodrigues committed:
fix: update linting and release-please

AngRodrigues committed:
ci: typos

AngRodrigues committed:
ci: keep release-please separate from linting

AngRodrigues committed:
ci: update testing

AngRodrigues committed:
ci: add pytest

AngRodrigues committed:
ci: update testing workflow

AngRodrigues committed:
ci: update linting action

AngRodrigues committed:
chore: upd conda meta

AngRodrigues committed:
style: style fixes by ruff and autoformatting by black

AngRodrigues committed:
ci: add conda & pypi

AngRodrigues committed:
Merge branch 'v3.2' of https://github.com/Loop3D/map2loop into v3.2

AngRodrigues committed:
ci: add docs

AngRodrigues committed:
ci: update to ubuntu24.04

AngRodrigues committed:
ci: force docs to branch gh ref

AngRodrigues committed:
chore: add testing badge to readme

AngRodrigues committed:
chore: add release badge

AngRodrigues committed:
Update README.md

AngRodrigues committed:
Update README.md

AngRodrigues committed:
fix: update conda and pip builds

AngRodrigues committed:
ci: add classic solver

AngRodrigues committed:
ci: add extra action to test conda build and install

AngRodrigues committed:
ci update

AngRodrigues committed:
ci: update

AngRodrigues committed:
gdal pin?

AngRodrigues committed:
add libmamba

AngRodrigues committed:
chore: typos

AngRodrigues committed:
Fix: update gdal install in actions (#163)

AngRodrigues committed:
ci: remove the build test for now

AngRodrigues committed:
Merge branch 'v3.2' of https://github.com/Loop3D/map2loop into v3.2
